crates/bevy_reflect/src/impls/std.rs

```diff
@@ -199,3 +204,70 @@ impl<K: Reflect + Clone + Eq + Hash, V: Reflect + Clone> Reflect for HashMap<K,
         None
     }
 }
+
+impl Reflect for Cow<'static, str> {
+    fn type_name(&self) -> &str {
+        std::any::type_name::<Self>()
+    }
+
+    fn any(&self) -> &dyn Any {
+        self
+    }
+
+    fn any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn apply(&mut self, value: &dyn Reflect) {
+        let value = value.any();
+        if let Some(value) = value.downcast_ref::<Self>() {
```

Comment:
We've removed type coercion from the reflect api (ex: we used to coerce all ints to the appropriate precision, but now we dont). I think for simplicity / consistency we should only accept `Cow<'static, str>` here unless absolutely necessary.

Reply:
✅ Coercion removed.